src/test/java/io/cdap/plugin/successfactors/source/transport/SuccessFactorsTransporterTest.java

```diff
@@ -0,0 +1,180 @@
+import okhttp3.OkHttpClient;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.net.HttpURLConnection;
+import java.net.SocketTimeoutException;
+import java.net.UnknownHostException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import javax.ws.rs.core.MediaType;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+
+public class SuccessFactorsTransporterTest {
+
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+  @Rule
+  public WireMockRule wireMockRule = new WireMockRule(wireMockConfig().dynamicPort().dynamicHttpsPort());
+  private SuccessFactorsPluginConfig.Builder pluginConfigBuilder;
+  private SuccessFactorsTransporter transporter;
+  private SuccessFactorsPluginConfig pluginConfig;
+  private SuccessFactorsUrlContainer successFactorsURL;
+
+  @BeforeClass
+  public static void classSetup() {
+    new MockUp<SuccessFactorsTransporter>() {
+      @Mock
+      public OkHttpClient.Builder getConfiguredClient() {
+        return allowAllSSL();
+      }
+    };
+  }
+
+  private static OkHttpClient.Builder allowAllSSL() {
+    TrustManager[] trustAllCerts = new TrustManager[]{
+      new X509TrustManager() {
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+          return new X509Certificate[]{};
+        }
+      }
+    };
+
+    try {
+      SSLContext trustAllSslContext = SSLContext.getInstance("SSL");
+      trustAllSslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+      SSLSocketFactory trustAllSslSocketFactory = trustAllSslContext.getSocketFactory();
+
+      HostnameVerifier customHostVerifier = new HostnameVerifier() {
+        @Override
+        public boolean verify(String hostname, SSLSession session) {
+          return true;
+        }
+      };
+
+      OkHttpClient.Builder builder = new OkHttpClient().newBuilder();
+      builder.sslSocketFactory(trustAllSslSocketFactory, (X509TrustManager) trustAllCerts[0]);
+      builder.hostnameVerifier(customHostVerifier);
+      return builder;
+    } catch (NoSuchAlgorithmException | KeyManagementException e) {
+      e.printStackTrace();
```

Comment:
use logger

Reply:
If these exceptions are not expected, then don't catch them - just fail the test

Reply:
But this is the BeforeClass setup, not the test case. Exceptions are needed here as getInstance() throws NoSuchAlgorithmException and init() throws KeyManagementException. Done the change for Logger.